### ISSUE TITLE
integrate into express app

### ISSUE THREAD

**Issue by martin0852** [jishi/node-sonos-http-api#429]
Is there any guidance on how I could integrate this into an exiting app? I am using ngrok and can navigate to http://host/zones just fine from a rest client. However I am not sure how to so the following:

Handle a post request to my app - OK
Do some logic and prepare a 'GET' request to the http://host/zones - OK
Send the get request to the http:/localhost:5000/zones from within express app running on localhost also (same port too) - NOK

The basic idea is to be able to control the SONOS from anywhere and I am unclear on how to handle this from the documentation.

If this question has been raised before please redirect me towards any info.

Thanks, M

**Comment by jishi** [comment]
Basically, you would just invoke the request, and then pipe the stream from the request onto the reguest handler. 

This is outside of the scope of this app, since there is no corner case for this, it's basically a question on how to proxy some routes in express to another host. Since you want to use POST instead of the normal GET pattern, you can't just blindly proxy it, but you should be able to just make a request (with native http module or the request library), and then just take that res object, and call `sonosResponseObject.pipe(expressResponseObject)`

In your example you use port 5000, which isn't default, but I guess that was just a typo.

**Comment by martin0852** [comment]
Hi,  thanks for that info - I realise it is outside the scope so thanks for taking the time to answer. I actually changed the port to 5000 as that's what I was using for my express app (I was trying to mash the two together). However what I ended up doing to get it working is partition the two systems and have the app talk to the Sonos controller with GET commands(as normal but instead of localhost I used the host ngrok gave me for this instance) I will try to use the suggestion you made and see how I get on. Thanks again for the response!
